--- Makefile
+++ Makefile	1998/11/09 11:51:57
@@ -24,12 +27,8 @@
 	$(RM) -f *.o core buffer .merrs
 
 install: buffer
-	rm -f $(INSTBIN)/buffer
-	cp buffer $(INSTBIN)/buffer
-	chmod 111 $(INSTBIN)/buffer
-	rm -f $(INSTMAN)/buffer.$S
-	cp buffer.man $(INSTMAN)/buffer.$S
-	chmod 444 $(INSTMAN)/buffer.$S
+	install -s buffer $(INSTBIN)/buffer
+	install -m 644 buffer.man $(INSTMAN)/buffer.$S
 
 buffer.tar: $(ALL)
 	$(RM) -f buffer.tar
--- Makefile.Linux
+++ Makefile.Linux	1998/11/09 11:51:57
@@ -0,0 +1,6 @@
+compile:
+	make
+
+install:
+	make install
+
