--- Makefile
+++ Makefile	1998/11/09 11:51:57
@@ -24,12 +27,8 @@
 	$(RM) -f *.o core buffer .merrs
 
 install: buffer
-	rm -f $(INSTBIN)/buffer
-	cp buffer $(INSTBIN)/buffer
-	chmod 111 $(INSTBIN)/buffer
-	rm -f $(INSTMAN)/buffer.$S
-	cp buffer.man $(INSTMAN)/buffer.$S
-	chmod 444 $(INSTMAN)/buffer.$S
+	install -s buffer $(INSTBIN)/buffer
+	install -m 644 buffer.man $(INSTMAN)/buffer.$S
 
 buffer.tar: $(ALL)
 	$(RM) -f buffer.tar
--- Makefile.Linux
+++ Makefile.Linux	1998/11/09 11:51:57
@@ -0,0 +1,6 @@
+compile:
+	make
+
+install:
+	make install
+
--- sem.c
+++ sem.c	1998/11/09 12:19:25
@@ -33,7 +33,7 @@
 #include <sys/sem.h>
 #include <errno.h>
 
-#if defined(SYS5) || defined(ultrix) || defined(_AIX)
+#if defined(SYS5) || defined(ultrix) || defined(_AIX) || defined _SEM_SEMUN_UNDEFINED
 union semun {
 	int val;
 	struct semid_ds *buf;
@@ -151,7 +151,7 @@
 	if( sem_id == -1 )
 		return;
 
-	if( semctl( sem_id, 0, IPC_RMID, NULL ) == -1 ){
+	if( semctl( sem_id, 0, IPC_RMID, (union semun)0 ) == -1 ){
 		report_proc();
 		perror( "internal error, failed to remove semaphore" );
 	}
